File: problems/remove_dup_sorted/remove_duplicate.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        unq_idx = 0
        for idx in range(1, len(nums)):
            if nums[idx] != nums[unq_idx]:
                unq_idx += 1
                nums[unq_idx] = nums[idx]
        return unq_idx + 1

File: problems/remove_dup_sorted/RemoveDuplicate.java
package problems.remove_dup_sorted;

public class RemoveDuplicate {
    public int removeDuplicates(int[] nums) {
        int uniqueIndex = 0;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] != nums[uniqueIndex]) {
                uniqueIndex += 1;
                nums[uniqueIndex] = nums[i];
            }
        }
        return uniqueIndex + 1;
    }
}
